javascript-client/src/client.ts

```diff
@@ -46,13 +46,44 @@ export interface ProgressEvent {
 
 type ProgressCallback = (progress: ProgressEvent) => void;
 
+export interface S3FFClientOptions {
+  readonly onProgress?: ProgressCallback;
+  readonly apiInstance?: AxiosInstance;
+  readonly apiConfig?: AxiosRequestConfig
+}
+
 export default class S3FFClient {
+  protected readonly api: AxiosInstance;
+
+  protected readonly onProgress: ProgressCallback;
+
+  /**
+   * Create an S3FFClient instance.
+   *
+   * @param baseUrl - The absolute URL to the Django server.
```

Comment:
In one case, where `apiInstance` / `apiConfig` has its `.baseUrl` property already set, the `baseUrl` here is technically optional and redundant.

I've thought about using factory methods, etc. to support that case, but all of it seems too complex for end users to easily figure out. I guess we'll force those users to just pass the information twice?

Reply:
Requiring it to be passed as an argument sounds like the simplest way to ensure that it's specified, even if it might be redundant.